lib/significant-other.js

```diff
@@ -7,35 +7,55 @@ import {
 
 export default {
   subscriptions: null,
+  cache: {},
   activate(state) {
     // Events subscribed to in atom's system can be easily cleaned up with a
     // CompositeDisposable
-    this.subscriptions = new CompositeDisposable();
+    this.subscriptions = new CompositeDisposable()
+    this.cache = state.cache || {}
 
     // Register command that toggles this view
     this.subscriptions.add(atom.commands.add('atom-text-editor', {
-      'significant-other:toggle': () => this.toggle()
+      'significant-other:toggle': () => this.toggle(),
+      'significant-other:clear-cache': () => this.clearCache()
     }));
   },
 
   deactivate() {
-    this.subscriptions.dispose();
+    this.subscriptions.dispose()
+    this.cache = {}
   },
 
   serialize() {
-    return {}
+    return {
+      cache: this.cache
+    }
   },
 
   getFilePath() {
     return atom.workspace.getActiveTextEditor().getPath()
   },
 
+  clearCache() {
+    this.cache = {}
+  },
+
   toggle() {
     const filePath = this.getFilePath()
     const matchmaker = Matchmaker.for(filePath)
 
+    // use cached path if we can
+    if (this.cache[filePath]) {
+      atom.workspace.open(this.cache[filePath])
```

Comment:
I don't think `existsSync` is the option, you probably will hit it too often compare to the false positive scenario when you try to oped deleted file.

I think that `existsSync` is expansive enough (as any IO) and we should avoid any IO operations as much as we can. 

We can try to clear the cache if an empty file was opened, but honestly speaking I thnk it's a bit  overkill.